storm-client/src/jvm/org/apache/storm/serialization/KryoTupleDeserializer.java

```diff
@@ -39,7 +38,7 @@ public KryoTupleDeserializer(final Map<String, Object> conf, final GeneralTopolo
         _kryoInput = new Input(1);
     }        
 
-    public Tuple deserialize(byte[] ser) {
+    public TupleImpl deserialize(byte[] ser) {
```

Comment:
It works because Java supports covariant return types. But I am not sure why is this changed since I don't see this being made use of to avoid the downcast in the patch.